packages/nexrender-core/src/helpers/tracer.js

```diff
@@ -0,0 +1,20 @@
+var tracer
+
+class NoopSpan {
+    setTag(_key, _value) { return this }
+}
+
+class NoopTracer {
+    wrap(_name, cb) { return cb() }
+
+    trace(_name, cb) { return cb(new NoopSpan()) }
+}
+
+if(process.env.ENABLE_DATADOG_APM) {
+    tracer = require('dd-trace').init();
```

Comment:
Now that we have this module to encapsulate whether we're using an actual Datadog client or not, perhaps it's also the time to fully abstract away `dd-trace` entirely and not even return the Tracer from `init()` but rather a thin facade over `dd-trace` that only implements NoopTracer and NoopSpan. e.g.

    class DatadogTracer {
      constructor() {
        this.tracer = require(`dd-trace`).init();
      }
      wrap(_name, cb) { return this.tracer.wrap(_name, cb); } // or whatever this is...
      trace(_name, cb) { ... } 
    }

    tracer = new DatadogTracer();

Reply:
Totally agree here, I'm going to add an issue and merge this so I can continue digging into deploys.